packages/sdk/src/mint/mint.ts

```diff
@@ -36,81 +35,125 @@ export const mint = (
     throw new Error(ERROR_MESSAGES.NO_MEDIA);
   }
 
-  if (splits) {
-    // FIXME: suggest we use % (ints) vs float here
-    // 0.5 -> 5000
-    adjustSplitsForContract(splits);
+  let adjustedRoyalties: Splits;
+  let royaltyTotal: number;
+  let roundedRoyalties: Splits;
+
+  //royalties adjustments to make devx better
+  if (royalties) {
+    royaltyTotal = getRoyaltyTotal(royalties);
+    adjustedRoyalties = adjustRoyaltiesForContract(royalties, royaltyTotal);
+    roundedRoyalties = roundRoyalties(adjustedRoyalties);
   }
 
-  if (splits && Object.keys(splits).length > 50) {
-    throw new Error(ERROR_MESSAGES.MAX_SPLITS);
+  //if royalties exist they need to be populated
+  if (royalties && Object.keys(royalties).length < 1) {
+    throw new Error(ERROR_MESSAGES.MIN_ROYALTIES);
   }
 
-  if (splits && Object.keys(splits).length < 2) {
-    throw new Error(ERROR_MESSAGES.SPLITS);
+  if (amount && (amount > 125 || amount < 1)) {
+    throw new Error(ERROR_MESSAGES.INVALID_AMOUNT);
   }
 
-  if (amount && amount > 125) {
-    throw  new Error(ERROR_MESSAGES.MAX_AMOUT);
+  //if specifying tokenIdsToMint these must be populated
+  if (tokenIdsToMint && tokenIdsToMint.length == 0) {
+    throw new Error(ERROR_MESSAGES.EMPTY_TOKEN_IDS);
   }
 
-  if (royaltyPercentage && royaltyPercentage < 0 || royaltyPercentage > 0.5) {
-    throw new Error(ERROR_MESSAGES.INVALID_ROYALTY_PERCENTAGE);
+  if (tokenIdsToMint && amount && tokenIdsToMint?.length !== amount) {
+    throw new Error(ERROR_MESSAGES.MUTUAL_EXCLUSIVE_AMOUNT);
   }
 
+  //if no amount or tokenids defined then default to 1 amount
+  const adjustedAmount = tokenIdsToMint?.length || amount || 1;
+
+
   return {
     contractAddress: contractAddress || mbjs.keys.contractAddress,
     args: {
       owner_id: ownerId,
       metadata: metadata,
-      num_to_mint: amount || 1,
+      num_to_mint: adjustedAmount,
       // 10_000 = 100% (see above note)
-      royalty_args: !splits ? null : { split_between: splits, percentage: royaltyPercentage * 10_000 },
-      split_owners: splits || null,
-      token_ids_to_mint: tokenIdsToMint,
+      royalty_args: !royaltyTotal ? null : { split_between: roundedRoyalties, percentage: Math.floor(royaltyTotal * 10000) },
```

Comment:
Any reason for flooring div over rounding?

Reply:
We can do round